src/widget.ts

```diff
@@ -724,52 +724,49 @@ export class DSVEditor extends Widget {
       update.selection = selection;
       // Add the command to the grid state.
       update.gridStateUpdate.nextCommand = command;
-      this.updateModel(update);
       this._grid.selectionModel.select(newSelection);
     }
+    this.updateModel(update);
   }
 
   /**
    * Updates the current transaction with the raw data, header, and changeArgs
    * @param update The modelChanged args for the Datagrid (may be null)
    */
   public updateModel(update?: DSVEditor.ModelChangedArgs): void {
-    // if not selection was passed through, take the current selection
-    // Bail early if there is no update.
-    if (!update) {
-      return;
-    }
-    // If no selection property was passed in, record the current selection.
-    // grab current selection if none exists
-    if (!update.selection) {
-      update.selection = this._grid.selectionModel.currentSelection();
-    }
-    // for every litestore change except the init, set the dirty boolean to true
-    this.dirty =
-      update &&
-      update.gridStateUpdate &&
-      update.gridStateUpdate.nextCommand === 'init'
-        ? false
-        : true;
-    // Update the litestore.
-    this._litestore.beginTransaction();
-    this._litestore.updateRecord(
-      {
-        schema: DSVEditor.DATAMODEL_SCHEMA,
-        record: DSVEditor.RECORD_ID
-      },
-      {
-        rowMap: update.rowUpdate || DSVEditor.NULL_NUM_SPLICE,
-        columnMap: update.columnUpdate || DSVEditor.NULL_NUM_SPLICE,
-        valueMap: update.valueUpdate || null,
-        selection: update.selection || null,
-        gridState: update.gridStateUpdate || null
+    if (update) {
+      // If no selection property was passed in, record the current selection.
+      if (!update.selection) {
+        update.selection = this._grid.selectionModel.currentSelection();
       }
-    );
+      // for every litestore change except the init, set the dirty boolean to true
+      this.dirty =
+        update &&
+        update.gridStateUpdate &&
+        update.gridStateUpdate.nextCommand === 'init'
+          ? false
+          : true;
+      // Update the litestore.
+      this._litestore.beginTransaction();
+      this._litestore.updateRecord(
+        {
+          schema: DSVEditor.DATAMODEL_SCHEMA,
+          record: DSVEditor.RECORD_ID
+        },
+        {
+          rowMap: update.rowUpdate || DSVEditor.NULL_NUM_SPLICE,
+          columnMap: update.columnUpdate || DSVEditor.NULL_NUM_SPLICE,
+          valueMap: update.valueUpdate || null,
+          selection: update.selection || null,
+          gridState: update.gridStateUpdate || null
+        }
+      );
+      this._litestore.endTransaction();
+    }
+
     if (this.dataModel.isDataFormatted) {
       this._updateRenderer();
     }
```

Comment:
Is it possible to move these lines to the top and still have the bail statement? We don't want to nest things inside a conditional if we don't need to.

Reply:
I think it breaks if we do that b/c updateRenders activates a repaint before we have updated the model but let me give it a try.

Reply:
Yeah it seems to not serialize correctly if we do it that way. We get undefined values. My guess is that we end up having renders finding incorrect data types and somehow that get's marked as undefined. Not totally sure and this is probably worth investigating further. The current version seems to work.